sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/ParDo.java

```diff
@@ -660,11 +659,25 @@ public static DoFnSchemaInformation getDoFnSchemaInformation(
      */
     public SingleOutput<InputT, OutputT> withSideInputs(
         Iterable<? extends PCollectionView<?>> sideInputs) {
+      Map<String, PCollectionView<?>> mappedInputs =
+          StreamSupport.stream(sideInputs.spliterator(), false)
+              .collect(Collectors.toMap(v -> v.getTagInternal().getId(), v -> v));
+      return withSideInputs(mappedInputs);
+    }
+
+    /**
+     * Returns a new {@link ParDo} {@link PTransform} that's like this {@link PTransform} but with
+     * the specified additional side inputs. Does not modify this {@link PTransform}.
+     *
+     * <p>See the discussion of Side Inputs above for more explanation.
+     */
+    public SingleOutput<InputT, OutputT> withSideInputs(
+        Map<String, PCollectionView<?>> sideInputs) {
       return new SingleOutput<>(
           fn,
-          ImmutableList.<PCollectionView<?>>builder()
-              .addAll(this.sideInputs)
-              .addAll(sideInputs)
+          ImmutableMap.<String, PCollectionView<?>>builder()
+              .putAll(this.sideInputs)
+              .putAll(sideInputs)
               .build(),
           fnDisplayData);
     }
```

Comment:
We should support builder-style as well. e.g.

ParDo....
.withSideInput("tag1", pcv1)
.withSideInput("tag2", pcv2);

Reply:
Done

Reply:
I don't see a withSideInput method?

Reply:
Added a new method `withSideInputs` which takes `tagId` , `PCollectionView` and internally calls the overloaded method which is taking map

Reply:
I don't see it. Are you sure it got pushed to github?

Reply:
https://github.com/apache/beam/pull/9275/files#diff-3e1c028dfd4b9b6b3e02849efa893c95R689